src/easydynamics/sample_model/sample_model.py

```diff
@@ -0,0 +1,277 @@
+import warnings
+from typing import List, Optional, Union
+
+import numpy as np
+import scipp as sc
+from easyscience.base_classes import CollectionBase
+from easyscience.global_object.undo_redo import NotarizedDict
+from easyscience.job.theoreticalmodel import TheoreticalModelBase
+
+from .components.model_component import ModelComponent
+
+Numeric = Union[float, int]
+
+
+class SampleModel(CollectionBase, TheoreticalModelBase):
+    """
+    A model of the scattering from a sample, combining multiple model components.
```

Comment:
Similarly, you could extend the API to include `__contains__` so we can do:
```python
>>> model = SampleModel()
>>> gaussian = Gaussian(name="g1")
>>> model.add_component(gaussian)
>>> "g1" in model
True
>>> gaussian in model
True
>>> "nonexistent" in model
False
```

with something like

```python
def __contains__(self, item: Union[str, ModelComponent]) -> bool:
    if isinstance(item, str):
        # Check by component name
        return any(comp.name == item for comp in self)
    elif isinstance(item, ModelComponent):
        # Check by component instance
        return any(comp is item for comp in self)
    else:
        return False
```